packages/@aws-cdk/aws-ec2/lib/machine-image.ts

```diff
@@ -363,22 +363,25 @@ export interface GenericWindowsImageProps {
  * manually specify an AMI map.
  */
 export class GenericLinuxImage implements IMachineImage {
-  constructor(private readonly amiMap: {[region: string]: string}, private readonly props: GenericLinuxImageProps = {}) {
-  }
-
-  public getImage(scope: Construct): MachineImageConfig {
-    const region = Stack.of(scope).region;
-    if (Token.isUnresolved(region)) {
-      throw new Error('Unable to determine AMI from AMI map since stack is region-agnostic');
-    }
+  private mapping: { [k1: string]: { [k2: string]: any } } = {};
 
-    const ami = region !== 'test-region' ? this.amiMap[region] : 'ami-12345';
-    if (!ami) {
-      throw new Error(`Unable to find AMI in AMI map: no AMI specified for region '${region}'`);
+  /**
+   * A Linux image where you specify the AMI ID for every region
+   *
+   * @param amiMap For every region where you are deploying the stack,
+   * specify the AMI ID for that region.
+   * @param props Customize the image by supplying additional props
+   */
+  constructor(readonly amiMap: { [region: string]: string }, private readonly props: GenericLinuxImageProps = {}) {
+    for (const [region, ami] of Object.entries(amiMap)) {
+      this.mapping[region] = { ami };
     }
+  }
 
+  public getImage(scope: Construct): MachineImageConfig {
+    const amiMap = new CfnMapping(scope, 'AmiMap', { mapping: this.mapping });
     return {
-      imageId: ami,
+      imageId: amiMap.findInMap(Aws.REGION, 'ami'),
```

Comment:
This needs to be using `Stack.of(scope.region)`, and I'd prefer it check if the value is `Token.isUnresolved()` before creating `CfnMapping`s. If not, we can just do a direct lookup like before.